src/modules/internet/index.ts

```diff
@@ -804,4 +805,104 @@ export class InternetModule extends ModuleBase {
       this.faker.definitions.internet.emoji[emojiType]
     );
   }
+
+  /**
+   * Generates a random JWT (JSON Web Token) Algorithm.
+   *
+   * @see faker.internet.jwt(): For generating random JWT (JSON Web Token).
+   *
+   * @example
+   * faker.internet.jwtAlgorithm() // 'HS256'
+   * faker.internet.jwtAlgorithm() // 'RS512'
+   *
+   * @since 9.0.0
+   */
+  jwtAlgorithm(): string {
+    return this.faker.helpers.arrayElement(
+      this.faker.definitions.internet.jwt_algorithm
+    );
+  }
+
+  /**
+   * Generates a random JWT (JSON Web Token).
+   *
+   * @param options The optional options object.
+   * @param options.header The Header to use for the token. Defaults to a random object with the following fields: `alg` and `typ`.
+   * @param options.payload The Payload to use for the token. Defaults to a random object with the following fields: `iat`, `exp`, `nbf`, `iss`, `sub`, `aud`, and `jti`.
+   * @param options.refDate The date to use as reference point for the newly generated date.
+   *
+   * @see https://datatracker.ietf.org/doc/html/rfc7519
+   * @see faker.internet.jwtAlgorithm(): For generating random JWT (JSON Web Token) Algorithm.
+   *
+   * @example
+   * faker.internet.jwt()
+   * faker.internet.jwt({ header: { alg: 'HS256' }}) // 'eyJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE3MTg2MTM3MTIsImV4cCI6MTcxODYzMzY3OSwibmJmIjoxNjk3MjYzNjMwLCJpc3MiOiJEb3lsZSBhbmQgU29ucyIsInN1YiI6IjYxYWRkYWFmLWY4MjktNDkzZS1iNTI1LTJjMGJkNjkzOTdjNyIsImF1ZCI6IjczNjcyMjVjLWIwMWMtNGE1My1hYzQyLTYwOWJkZmI1MzBiOCIsImp0aSI6IjU2Y2ZkZjAxLWRhMzMtNGUxNi04MzJiLTFlYTk3ZGY1MTQ2YSJ9.5iUgaCaFVPZ8d1QD0xMjoeJbmPVyUfKfoRQ6Njzm5MLp5F4UMh5REbPCrW70fAkr'
+   * faker.internet.jwt({ payload: { iss: 'Acme' }}) // 'eyJhbGciOiJFUzM4NCIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJBY21lIn0.syUt0GBukNac8Cn1AGKFq2SWAXWy1YIfl0uOYiwg6TZ3omAW0c7FGWY6bC7ZOFSt'
+   * faker.internet.jwt({ refDate: '2020-01-01T00:00:00.000Z' }) // 'eyJhbGciOiJFUzM4NCIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1Nzc4MDY4NDUsImV4cCI6MTU3Nzg0NjI4MCwibmJmIjoxNTgxNTQyMDYwLCJpc3MiOiJLcmVpZ2VyLCBBbHRlbndlcnRoIGFuZCBQYXVjZWsiLCJzdWIiOiI5NzVjMjMyOS02MDlhLTRjYTYtYjBkZi05ZmY4MGZiNDUwN2QiLCJhdWQiOiI0ODQxZWYwNi01OWYwLTQzMWEtYmFmZi0xMjkxZmRhZDdhNjgiLCJqdGkiOiJmNDBjZTJiYi00ZWYyLTQ1MjMtOGIxMy1kN2Q4NTA5N2M2ZTUifQ.cuClEZQ0CyPIMVS5uxrMwWXz0wcqFFdt0oNne3PMryyly0jghkxVurss2TapMC3C'
+   *
+   * @since 9.0.0
+   */
+  jwt(
+    options: {
+      /**
+       * The header to use for the token. If present, it will replace any default values.
+       *
+       * @default
+       * {
+       *    alg: faker.internet.jwtAlgorithm(),
+       *    typ: 'JWT',
+       * };
```

Comment:
Usually we don't have entire objects as defautl values. Thats why I have another question regarding the current visualization in the documentation:

![image](https://github.com/faker-js/faker/assets/77127505/2a2aade8-ec41-4004-b16d-29572113e807)

While the default values in the displayed table are technically correct, they may appear unusual in the given context. Specifically, the dangling comma at the last property and the semicolon at the end stand out. No other value uses semicolons, which is why I find it somewhat unconventional. I#m not quite sure whether this is relevant, so what are your thoughts on this?

_The same issue applies to the payload property._

Reply:
Should we introduce jwtHeader and jwtPayload methods? I dont like that as it is a dynamic complex object.

Alternatively we could just hide the specifics of the default e.g. `{ ... }`. And put the actual property names in the description. 🤔

Reply:
> Should we introduce jwtHeader and jwtPayload methods?

I don't see how this is related to my question. 

> Alternatively we could just hide the specifics of the default e.g. { ... }.

I don't think thats necessary. I'm not bothered by the actual default value itself, but the formatting. Simply removing the dangling comma and semicolon would already solve this (unless the documentation generation uses different formatting). Like this:

```suggestion
       * {
       *    alg: faker.internet.jwtAlgorithm(),
       *    typ: 'JWT'
       * }
```

My question was whether anyone else is botheres by that problem, or if I'm beeing nitpicky.

Reply:
> > Should we introduce jwtHeader and jwtPayload methods?
> 
> I don't see how this is related to my question.

Instead of defining the whole object in there, we would simply refer to the method:

````ts
 * @default faker.internet.jwtHeader()
````

> My question was whether anyone else is bothered by that problem, or if I'm being nitpicky.

I like your suggested `@default` better than the representation used by this PR, even if prettier might disagree.

1) Should we handle this case on the script side instead? E.g. `defaultValue.replaceAll(/,(\n *)}/g, '$1}')`

The trailing `;` should be removed in the code as well (as we don't have that anywhere else either).

2) And should we also use `defaultValue.replaceAll(/: ([a-z0-9"'-])/g, ': $1')` (with nbsp)?

**This PR**
![grafik](https://github.com/faker-js/faker/assets/1579362/b86941c1-33b3-4fad-9120-2c9de7bd9d0b)

**With script optimizations**
![grafik](https://github.com/faker-js/faker/assets/1579362/37a3010d-9034-4034-ac0e-84fd40fcabb2)

---

I just noticed that the default value uses an indentation of 3 spaces instead of prettier's 2.

**For script change**
```suggestion
       * {
       *   alg: faker.internet.jwtAlgorithm(),
       *   typ: 'JWT',
       * }
```

**For source code change**
```suggestion
       * {
       *   alg: faker.internet.jwtAlgorithm(),
       *   typ: 'JWT'
       * }
```

Reply:
I created https://github.com/faker-js/faker/pull/3190 to improve the visualization of the default value.